tests/ssl-opt.sh

```diff
@@ -656,14 +656,32 @@ fi
 # used by watchdog
 MAIN_PID="$$"
 
-# be more patient with valgrind
+# We use somewhat arbitrary delays for tests:
+# - how long do we wait for the server to start (when lsof not available)?
+# - how long do we wait for the client to finish?
+#   (not to check performance, just to avoid waiting indefinitely)
+# - how long do we wait after client exit for completion of server tasks?
+#   (in particular, for it to finish writing & flushing its log file)
+# Things are slower with valgrind, so give extra time here.
+#
+# Note: there is a trade-off between the running time of this script and the
+# risk of spurious errors because we didn't wait long enough. In a better
+# world, we would be able to do without those delays (except the watchdog).
 if [ "$MEMCHECK" -gt 0 ]; then
-    START_DELAY=3
-    DOG_DELAY=30
+    START_DELAY=6
+    DOG_DELAY=60
+    SRV_DELAY_BASE=2
 else
-    START_DELAY=1
-    DOG_DELAY=10
+    START_DELAY=2
+    DOG_DELAY=15
+    # allow impatient devs to override this one (really impacts running time)
+    : ${SRV_DELAY_BASE:=1}
```

Comment:
Impatient devs are likely to have lsof. What's the impact when impatient devs have lsof?

Reply:
Unless I'm missing something, lsof only saves us from `$START_DELAY`, not from `$SRV_DELAY_BASE`: that one is intended to give the server some time to finish its thing after the client disconnected, including printing all the post-connection messages, and I don't think there is an easy way to check it has, so here we're just really waiting what we hope is a sufficient amount, and so far we have no better alternative in the script.

So actually without lsof you have a minimum of `$START_DELAY + $SRV_DELAY_BASE` spent just sleeping on each test; with lsof it's `epsilon + $SRV_DELAY_BASE`, so that's the case where `$SRV_DELAY_BASE` is the dominant one and you really want to make its value lower.

Reply:
OK. Sounds good to make this consistently configurable (valgrind or not). I can imagine a developer getting annoyed that his SRV_DELAY_BASE isn't being obeyed when he expects it to.